### PR TITLE
[language] use short_str_lossless in ModuleId printing

### PR DESCRIPTION
**Pull request by sblackshear**
This makes printing anything with a ModuleId much prettier, but without losing information.

### CODE REVIEW

**Review by runtian-zhou — APPROVED** [review]
Why not :)

**Comment by tnowacki** [comment]
Do you have a before/after example?

**Comment by sblackshear** [comment]
This was inspired by looking at the expected value tests in https://github.com/diem/diem/pull/7234. For example, something like 

```
Publishing a new module 00000000000000000000000000000042::M
Publishing a new module 00000000000000000000000000000043::N
Command `view storage/0x00000000000000000000000000000042/modules/M.mv`:
```

from there will become

```
Publishing a new module 0x42::M
Publishing a new module 0x43::N
Command `view storage/0x42/modules/M.mv`:
```

I expect there will be lots of other expected value tests I'll have to fix before landing this...

File: json-rpc/tests/integration_test.rs [hunk 1]
```diff
@@ -411,7 +411,7 @@ fn create_test_cases() -> Vec<Test> {
                             "reason": "EINSUFFICIENT_BALANCE",
                             "reason_description": " The account does not hold a large enough balance in the specified currency"
                         },
-                        "location": "00000000000000000000000000000001::DiemAccount",
+                        "location": "0x1::DiemAccount",
```
Comment (sblackshear) [inline json-rpc/tests/integration_test.rs:414]: @xli: Just checking that this is ok from a json-rpc perspective. If clients are expecting to parse the address from this strong, I can change the printing back to the more verbose mode.

**Comment by bob-wilson** [comment]
I remember that stuff broke when we changed to print uppercase hex (#6296). Any similar risk here?

**Comment by sblackshear** [comment]
Good question! That PR changed printing of `AccountAddress`, which is very broadly used. This one is changing the printing of `ModuleId`, which is only used in language tooling and VM error messages (which get threaded through json-rpc). I am fairly confident about the former, but I asked @xli about the latter.

File: json-rpc/tests/integration_test.rs (hunk 1 above)
Comment (xli) [reply json-rpc/tests/integration_test.rs:414]: It is fine

**Comment by davidiw** [comment]
Old PR, feel free to re-open.

**Comment by bob-wilson** [comment]
It's not clear why this was dropped. I hadn't realized that it never landed. @sblackshear Do you still think we should do this? I really like the more readable output, as long as it isn't disruptive to switch.

**Comment by sblackshear** [comment]
I would still like to see this go in; IIRC it was breaking some tests in ways that I didn't understand.

**Comment by bob-wilson** [comment]
Ah, OK. Feel free to close it for now if you don't have time. Or we could add it to our list of starter tasks?

**Comment by sblackshear** [comment]
Good idea, I'll close/create a task.

**Pull request closed without merging**